src/main/java/org/mifos/connector/ams/pesacore/camel/route/PesaRouteBuilder.java

```diff
@@ -130,7 +130,7 @@ public void configure() {
                     String mpesaReceiptNumber = exchange.getProperty(SERVER_TRANSACTION_RECEIPT_NUMBER, String.class);
 
                     PesacoreRequestDTO confirmationRequestDTO = getPesacoreDtoFromChannelRequest(channelRequest,
-                            transactionId);
+                            mpesaReceiptNumber);
```

Comment:
Since validation / partyLookup uses transactionId, shouldn't getPesacoreDtoFromChannelRequest continue using transactionId instead of mpesaReceiptNumber?